lib/times-table.js

```diff
@@ -0,0 +1,17 @@
+function timesTable(number) {
```

Comment:
this function is looking pretty good. it has a lot of stuff going on though - probably because this is a complicated function.

my biggest suggestion here would be to break this up into smaller functions. for example, you create the first row before creating the rest of the rows since the first row doesn't require multiplication. we could pull that out into a smaller function.

i don't see any huge improvements to be made to your logic besides potentially creating a recursive function since you're essentially doing the same logic on each line. recursion is a bit slower though than iteration.